### ISSUE TITLE
Rollback Node rename 

### ISSUE THREAD

**Issue by nicolaeschbach** [cyber-geiger/toolbox-storage#31]
**Describe the Issue**
Right now a node can be renamed with the rename() method on a StorageController. If the rename should be reverted it is not possible because the rename() marks all Nodes to be renamed as tombstone and creates new one with the new name. Thereby a node can't be renamed back. _(StorageException:Node ":Enterprise" does not exist (but exited in the past; It is a tombstone now))_ 
It is only possibly to revoke the renamed Nodes by reading the oldNodes with getNodeOrTombstone() from the DB change the tombstone field to false and update them with the controller.
To do a optimistic replication of the DB it would be better if there is a more logical way to do this.
Or is the behavior extra designed like this and shouldn't be changed?

**Environmental information**
- SKD 2.14.4
-  geiger_localstorage: ^0.6.15

**Expected behavior**
Possibility to rollback a rename, without doing it manually.

**Comment by martingwerder** [comment]
This is possibly related to issue #28. I will investigate and publish a fix today.

**Comment by martingwerder** [comment]
Fix is in 0.6.17 can anyone confirm?

**Comment by manuGe** [comment]
The rename rollback is working now. However, the "rename by name" function does no longer function. The following code produces a `StorageException:Node ":Enterprise2" not found` exception.

```dart
await controller1.rename(':Enterprise', 'Enterprise2');
print((await controller1.get(':Enterprise2')).name);
```

**Comment by martingwerder** [comment]
Confirmed when using DummyMapper and renaming a root node... no matter how much many tests you write: You always miss a case. 

Should be working with the newly pushed 0.6.18 and should be covered for future releases by a test for all ```StorageMapper```s.

**Comment by nicolaeschbach** [comment]
This issue seems to be solved now with 0.6.18.
